api/src/main/java/org/openmrs/Allergy.java

```diff
@@ -15,26 +15,54 @@
 import java.util.List;
 import java.util.UUID;
 
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.OrderBy;
+import javax.persistence.Table;
+
 import org.apache.commons.lang3.StringUtils;
 import org.openmrs.util.OpenmrsUtil;
 
 /**
  * Represent allergy
  */
+@Entity
+@Table(name = "allergy")
 public class Allergy extends BaseChangeableOpenmrsData {
 	
 	public static final long serialVersionUID = 1;
 	
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
```

Comment:
AUTO is the default, so you can omit the strategy attribute

Reply:
i have fixed it